### ISSUE TITLE
Training on new data

### ISSUE THREAD

**Issue by gcelano** [360er0/COMBO#1]
I have a few questions as to the training of COMBO with my non-UD Latin data (from which the UD la_perseus data derives):

1) The format of my morphological annotation is always a 9-character string, such as a-p---nn-, where each character corresponds to a morphological feature (with the first one being the POS). Should I format the morphological features as -|p|-|-|-|n|n|- (while the first character "a" occupies the UPOS position in the CoNLL-U format)?

2) I see that, when using your la_perseus model, the XPOS field is not predicted: is there a reason for that? 

2) My treebank is a dependency treebank: the annotation scheme is similar to, but not the same as, the UD one: is COMBO indifferent to such distinctions?

1) Is the parameter "--embed external_embedding.txt" necessary? (Did you know whether it was used for la_perseus in the Shared Task 2018?) 

2) Can COMBO also perform automatic evaluation against the test data provided?

**Comment by 360er0** [comment]
Ad 1. COMBO assumes that morphological feature will be a list of "key=value" pairs. So I would suggest using the following format: key1=-|key2=p|key3=-|key4=-|key5=-|key6=n|key7=n|key8=-

Ad 2. The only reason is that during the Shared Task 2018 this field was not required. If you would like to train your own model with XPOS add the "xpostag" to the "targets" argument and appropriate value to "loss_weights". E.g. "--targets head,deprel,lemma,upostag,xpostag,feats --loss_weights 0.2,0.8,0.05,0.05,0.05,0.2"

Ad 3. Yes, as long as the format of the data is the same as UD COMBO should work.

Ad 4. No, it is not necessary, however during the Shared Task 2018 [this](https://github.com/facebookresearch/fastText/blob/master/pretrained-vectors.md) embeddings were used.

Ad 5. Yes, use "--mode evaluate --model model.pkl --test test.conllu".

**Comment by gcelano** [comment]
Thanks! I'll try that.

**Comment by gcelano** [comment]
I launched the training, but I had to stop it after 10 hours (at "Train iter 25"). I have less than 10000 tokens and a MacBook Pro (2.8 GHz, Intel Core i7): do you know how long the training phase can approximately last?

**Comment by 360er0** [comment]
For 10k sentences it should take about 3 hours to train on GPU, however on CPU it is much slower (~10x times). If you don't have access to machine with decent GPU I would suggest to limit the number of training epochs ("--epochs 100"). Each "training iter" consists of 5 epochs, so it should take you around 8 hours to train it. Since the training is shorter it will result in lower accuracy, but the difference is usually less than 1 p.p.

**Comment by gcelano** [comment]
Hi, I have been able to  successfully train my data. I see that the accuracies I get are lower than the ones for the almost corresponding UD data. Did you use word embeddings for Latin?

**Comment by 360er0** [comment]
Yes, I have used fastText embeddings.

**Comment by gcelano** [comment]
the one available here?:

https://fasttext.cc/docs/en/crawl-vectors.html

I guess that one can use it in COMBO via --embed=FILE. Did you use the bin or text ones?

**Comment by 360er0** [comment]
Yes. The text version.
